src/edu/ucsb/nceas/metacat/dataone/MNodeService.java

```diff
@@ -2204,35 +2204,7 @@ public Identifier generateIdentifier(Session session, String scheme, String frag
     @Override
     public QueryEngineDescription getQueryEngineDescription(Session session, String engine)
         throws InvalidToken, ServiceFailure, NotAuthorized, NotImplemented, NotFound {
-        if (engine != null && engine.equals(EnabledQueryEngines.PATHQUERYENGINE)) {
-            if (!EnabledQueryEngines.getInstance().isEnabled(EnabledQueryEngines.PATHQUERYENGINE)) {
-                throw new NotImplemented("0000", "MNodeService.query - the query engine " + engine
-                    + " hasn't been implemented or has been disabled.");
-            }
-            QueryEngineDescription qed = new QueryEngineDescription();
-            qed.setName(EnabledQueryEngines.PATHQUERYENGINE);
-            qed.setQueryEngineVersion("1.0");
-            qed.addAdditionalInfo("This is the traditional structured query for Metacat");
-            Vector<String> pathsForIndexing = null;
-            try {
-                pathsForIndexing = SystemUtil.getPathsForIndexing();
-            } catch (MetacatUtilException e) {
-                logMetacat.warn("Could not get index paths", e);
-            }
-            for (String fieldName : pathsForIndexing) {
-                QueryField field = new QueryField();
-                field.addDescription("Indexed field for path '" + fieldName + "'");
-                field.setName(fieldName);
-                field.setReturnable(true);
-                field.setSearchable(true);
-                field.setSortable(false);
-                // TODO: determine type and multivaluedness
-                field.setType(String.class.getName());
-                //field.setMultivalued(true);
-                qed.addQueryField(field);
-            }
-            return qed;
-        } else if (engine != null && engine.equals(EnabledQueryEngines.SOLRENGINE)) {
+        if (engine != null && engine.equals(EnabledQueryEngines.SOLRENGINE)) {
```

Comment:
if we only support solr, now, does it make sense to simplify the code by making it solr specific, and removing the property altogether? 

We could also remove `EnabledQueryEngines`?

(Even if we move away from solr in future, it would not be as simple as changing the property - we'd still need to add code to support the new engine)

Reply:
The DataONE API explicitly supports multiple query engines, and so we need to maintain this for API compatibility here. Plus, I suspect we could soon add another SOLR query engine for the entity search or other SOLR schemas.